### DISCUSSION TITLE
How to process ALTER statements with default value

### DISCUSSION THREAD

**Discussion by humada05** [JSQLParser/JSqlParser#1349] [Q&A]
Hi, I'm using JSqlParser for reading ALTER statements that come from DDL logs. Specifically, Im mainly interested in finding out whether the ALTER statement has a default value and what that value is exactly.

ex.
`ALTER TABLE SAMP.EMP_ACT ADD COLUMN DEPTNO INT DEFAULT 1`

From my basic testing, im able to retrieve the default values but only after combing through two layers of lists within the Alter class that is returned.
<img width="929" alt="Screen Shot 2021-09-24 at 4 54 46 AM" src="https://user-images.githubusercontent.com/16727862/134606541-7b3713c1-d4c1-4372-b4b8-72bd4fdba76d.png">

Is there a neater way to get this information without having to call `.getAlterExpressions().get(0).getColDataTypeList().get(0)...`

**Comment by wumpz** [comment]
Unfortunately, the parsing of **alter** statements especially those parameters is done in a very unstructured way. As you already found out the parameters are stored within a list.
